tests/utils/sampler_test.py

```diff
@@ -79,42 +78,41 @@ def test_dirichlet_sampler_with_assigned_probability() -> None:
     # Set the random seeds
     set_all_random_seeds(2023)
     sampler = DirichletLabelBasedSampler(unique_labels=list(range(10)), sample_percentage=1.0, beta=0.1)
-    ds = MnistDataset(data_path=Path("examples/datasets/MNIST"), train=True)
 
     # No heterogeneity
-    relative_probs = [1] * 10
-    sampler.probabilities = np.array([i for i in relative_probs]) / sum([i for i in relative_probs])
+    relative_probabilities = [1] * 10
+    sampler.probabilities = np.array([i for i in relative_probabilities]) / sum([i for i in relative_probabilities])
 
-    new_ds = sampler.subsample(ds)
+    new_ds = sampler.subsample(SYNTHETIC_DATASET)
     new_samples_per_class = [new_ds.targets[new_ds.targets == i].size(0) for i in range(10)]
-    new_probs = [x / sum(new_samples_per_class) for x in new_samples_per_class]
+    new_probabilities = [x / sum(new_samples_per_class) for x in new_samples_per_class]
 
-    # Assert that probabilities of sampled indicies are close to assigned probabilities
+    # Assert that probabilities of sampled indices are close to assigned probabilities
     for i in range(10):
-        assert pytest.approx(new_probs[i], abs=0.001) == sampler.probabilities[i]
+        assert pytest.approx(new_probabilities[i], abs=0.001) == sampler.probabilities[i]
 
     # Mild heterogeneity
-    relative_probs = [i for i in range(1, 11)]
-    sampler.probabilities = np.array([i for i in relative_probs]) / sum([i for i in relative_probs])
+    relative_probabilities = [i for i in range(1, 11)]
+    sampler.probabilities = np.array([i for i in relative_probabilities]) / sum([i for i in relative_probabilities])
 
-    new_ds = sampler.subsample(ds)
+    new_ds = sampler.subsample(SYNTHETIC_DATASET)
     new_samples_per_class = [new_ds.targets[new_ds.targets == i].size(0) for i in range(10)]
-    new_probs = [x / sum(new_samples_per_class) for x in new_samples_per_class]
+    new_probabilities = [x / sum(new_samples_per_class) for x in new_samples_per_class]
 
-    # Assert that probabilities of sampled indicies are close to assigned probabilities
+    # Assert that probabilities of sampled indices are close to assigned probabilities
     for i in range(10):
-        assert pytest.approx(new_probs[i], abs=0.001) == sampler.probabilities[i]
+        assert pytest.approx(new_probabilities[i], abs=0.01) == sampler.probabilities[i]
```

Comment:
The synthetic dataset is a about 6x smaller than MNIST, so we saw a slight increase in statistical sampling fluctuation.